dspace/modules/api/src/main/java/org/datadryad/anywhere/AssociationAnywhere.java

```diff
@@ -0,0 +1,409 @@
+                {
+                    throw new AssociationAnywhereException(get.getResponseBodyAsString());
+                }
+            }
+
+            return status;
+        }
+        catch (Exception e)
+        {
+            throw new AssociationAnywhereException(e.getMessage());
+        }
+
+    }
+
+    /**
+     * Iterate over all Concepts and update credits available.
+     *
+     * @param context
+     * @throws Exception
+     */
+    private static void updateConcept(Context context) throws Exception {
+
+        Scheme scheme = Scheme.findByIdentifier(context,ConfigurationManager.getProperty("solrauthority.searchscheme.prism_publicationName"));
+        for(Concept concept : scheme.getConcepts())
+        {
+            for(AuthorityMetadataValue value : concept.getMetadata("internal", "journal", "customerId",AuthorityMetadataValue.ANY))
+            {
+                updateConcept(concept, value.getValue());
+            }
+        }
+
+        context.commit();
+    }
+
+    /**
+     * Update Credit for a specific customer id.
+     * @param context
+     * @param customerId
+     * @throws Exception
+     */
+    private static void updateConcept(Context context, String customerId)
+            throws Exception {
+
+        Scheme scheme = Scheme.findByIdentifier(context,ConfigurationManager.getProperty("solrauthority.searchscheme.prism_publicationName"));
+        for(Concept concept : scheme.getConcepts())
+        {
+            for(AuthorityMetadataValue value : concept.getMetadata("internal", "journal", "customerId",AuthorityMetadataValue.ANY))
+            {
+                if(customerId != null && customerId.equals(value.getValue()))
+                    updateConcept(concept, value.getValue());
+            }
+        }
+
+        context.commit();
+
+    }
+
+    /**
+     * Update Credit for a specific customer id.
+     * @param customerId
+     * @throws Exception
+     */
+    private static void updateConcept(Concept concept, String customerId)
+            throws Exception {
+
+
+        //internal.journal.integrated
+        //internal.journal.sponsorName
+        //internal.journal.subscriptionPaid
+        //internal.journal.paymentPlanType
+
+        if(customerId==null)
+        {
+            throw new AssociationAnywhereException("customerId cannot be null");
+        }
+
+        String credit = getCredit(customerId);
+        Document customerInfo = loadCustomerInfo(customerId);
+        String result = printDocument(customerInfo);
+
+        String customerName = getStringValue(customerInfo, "//displayName");
+        String classSubclassDescr = getStringValue(customerInfo, "//classSubclassDescr");
+        String statusCode = getStringValue(customerInfo, "//statusCode");
+
+        //set integrated, make sure integrated is true
+        changeConceptMetadataValue(concept,"customerId",customerId);
+        changeConceptMetadataValue(concept,"integrated","true");
```

Comment:
Do we have a field in association anywhere that can be leveraged for "integrated" or would you like to set it manually via Concept metadata?